src/EnergyPlus/DXCoils.cc

```diff
@@ -14655,140 +14655,111 @@ void ReportDXCoil(EnergyPlusData &state, int const DXCoilNum) // number of the c
+        dxCoil.LatCoolingEnergy = dxCoil.TotalCoolingEnergy - dxCoil.SensCoolingEnergy;
+        dxCoil.ElecCoolingConsumption = dxCoil.ElecCoolingPower * ReportingConstant;
+        dxCoil.CrankcaseHeaterConsumption = dxCoil.CrankcaseHeaterPower * ReportingConstant;
+        state.dataHVACGlobal->DXElecCoolingPower = dxCoil.ElecCoolingPower;
+        dxCoil.EvapCondPumpElecConsumption = dxCoil.EvapCondPumpElecPower * ReportingConstant;
+        dxCoil.EvapWaterConsump = dxCoil.EvapWaterConsumpRate * ReportingConstant;
+        if (any_eq(dxCoil.CondenserType, DataHeatBalance::RefrigCondenserType::Evap)) {
+            dxCoil.BasinHeaterConsumption = dxCoil.BasinHeaterPower * ReportingConstant;
         }
     } break;
     }
 
-    if (state.dataDXCoils->DXCoil(DXCoilNum).CondensateCollectMode == CondensateCollectAction::ToTank) {
+    if (dxCoil.CondensateCollectMode == CondensateCollectAction::ToTank) {
         // calculate and report condensation rates  (how much water extracted from the air stream)
         // water flow of water in m3/s for water system interactions
         //  put here to catch all types of DX coils
-        Tavg = (state.dataDXCoils->DXCoil(DXCoilNum).InletAirTemp - state.dataDXCoils->DXCoil(DXCoilNum).OutletAirTemp) / 2.0;
-        RhoWater = RhoH2O(Tavg);
-        // CR9155 Remove specific humidity calculations
-        SpecHumIn = state.dataDXCoils->DXCoil(DXCoilNum).InletAirHumRat;
-        SpecHumOut = state.dataDXCoils->DXCoil(DXCoilNum).OutletAirHumRat;
+        Real64 Tavg = (dxCoil.InletAirTemp - dxCoil.OutletAirTemp) / 2.0;
         //  mdot * del HumRat / rho water
-        state.dataDXCoils->DXCoil(DXCoilNum).CondensateVdot =
-            max(0.0, (state.dataDXCoils->DXCoil(DXCoilNum).InletAirMassFlowRate * (SpecHumIn - SpecHumOut) / RhoWater));
-        state.dataDXCoils->DXCoil(DXCoilNum).CondensateVol = state.dataDXCoils->DXCoil(DXCoilNum).CondensateVdot * ReportingConstant;
-
-        state.dataWaterData->WaterStorage(state.dataDXCoils->DXCoil(DXCoilNum).CondensateTankID)
-            .VdotAvailSupply(state.dataDXCoils->DXCoil(DXCoilNum).CondensateTankSupplyARRID) = state.dataDXCoils->DXCoil(DXCoilNum).CondensateVdot;
-        state.dataWaterData->WaterStorage(state.dataDXCoils->DXCoil(DXCoilNum).CondensateTankID)
-            .TwaterSupply(state.dataDXCoils->DXCoil(DXCoilNum).CondensateTankSupplyARRID) = state.dataDXCoils->DXCoil(DXCoilNum).OutletAirTemp;
+        dxCoil.CondensateVdot =
+            max(0.0, (dxCoil.InletAirMassFlowRate * (dxCoil.InletAirHumRat - dxCoil.OutletAirHumRat) / Psychrometrics::RhoH2O(Tavg)));
+        dxCoil.CondensateVol = dxCoil.CondensateVdot * ReportingConstant;
+
+        state.dataWaterData->WaterStorage(dxCoil.CondensateTankID).VdotAvailSupply(dxCoil.CondensateTankSupplyARRID) = dxCoil.CondensateVdot;
+        state.dataWaterData->WaterStorage(dxCoil.CondensateTankID).TwaterSupply(dxCoil.CondensateTankSupplyARRID) = dxCoil.OutletAirTemp;
```

Comment:
Used the `dxcoil` member variables `dxCoil.InletAirHumRat` and `dxCoil.OutletAirHumRat`, instead of local variables.